contacts/app.rb

```diff
@@ -0,0 +1,69 @@
+require 'sinatra'
+require 'sinatra/reloader'
+require 'sinatra/activerecord'
+
+set :database, {adapter:'sqlite3', database:'database.db'}
+
+class Contact < ActiveRecord::Base
+end
+
+#homepage
+get '/' do
+    @contacts=Contact.all
+    erb :landing 
+end
+
+#search
+get '/search' do
+    query = "%#{params[:query]}%"
+    @contacts=Contact.where("first_name LIKE ? OR last_name LIKE ?", query, query)
+    erb :search
+end
+
+#adding new contacts
+get '/contacts/new' do
+    erb :new
+end
+
+#synching new contacts to database
+post '/contacts' do
+    contact=Contact.new
+    contact.first_name=params[:first_name]
+    contact.last_name=params[:last_name]
+    contact.phone=params[:phone]
+    contact.email=params[:email]
+    contact.address=params[:address]
+    contact.city=params[:city]
+    contact.state=params[:state]
+    contact.zip=params[:zip]
+    contact.save
+    redirect '/'
+end    
+
+#editing existing questions
+get '/contacts/edit' do
+    @contact = Contact.find(params[:id])
+    erb :edit
+end
+
+#synching edited contacts
+post '/contacts/edit' do
+    contact=Contact.find_by(params[:id])
+    contact.first_name=params[:first_name]
+    contact.last_name=params[:last_name]
+    contact.phone=params[:phone]
+    contact.email=params[:email]
+    contact.address=params[:address]
+    contact.city=params[:city]
+    contact.state=params[:state]
+    contact.zip=params[:zip]
+    contact.save
+    redirect "/"
+end
```

Comment:
In Sinatra, it will leave a blank field, but there is not a nil error thrown. I was noticing when working on a Rails app that when the placeholder attribute was used in the Edit form, if the fields you did not want to change are not re-filled, they get deleted. However, in Sinatra the untouched values were not changed or deleted. So, I'll have to try the value attribute for the Rails app.

Reply:
Also, I'll look into the PUT and PATCH verbs. We have not yet covered those in class. When you say resource, are you referring to the data we are looking back to from the database?